R/zchunk_L231.proc_sector.R

```diff
@@ -0,0 +1,460 @@
+      # Calculate service as energy * efficiency
+      mutate(service = value * efficiency) %>%
+      na.omit() %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(ind_output = sum(service)) %>%
+      ungroup() %>%
+      mutate(ind_proc_input = 0.008,
+             coefficient = ind_proc_input / ind_output,
+             supplysector = "industry",
+             subsector = "industry",
+             technology = "industry",
+             minicam.energy.input = "industrial processes") %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID")
+
+    # Interpolate coefficients to model years
+    L231.IndCoef <- L231.IndCoef %>%
+      select(-year, -coefficient, -ind_output, -ind_proc_input) %>%
+      distinct %>%
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      left_join(L231.IndCoef, by = c("GCAM_region_ID","region", "supplysector", "subsector", "technology", "minicam.energy.input", "year")) %>%
+      group_by(region, technology) %>%
+      mutate(coefficient = approx_fun(year, coefficient, rule = 2)) %>%
+      ungroup() %>%
+      select(region, supplysector, subsector, technology, year, minicam.energy.input, coefficient)
+
+    # ===================================================
+
+    # Produce outputs
+    L231.UnlimitRsrc %>%
+    add_title("Processing Resource Capacity Factors") %>%
+    add_units("Unitless") %>%
+    add_comments("Data from A31.rsrc_info") %>%
+    add_legacy_name("L231.UnlimitRsrc") %>%
+    add_precursors("emissions/A31.rsrc_info") ->
+    L231.UnlimitRsrc
+
+    L231.UnlimitRsrcPrice %>%
+    add_title("Processing Resource Prices") %>%
+    add_units("units") %>%
+    add_comments("Interpolated data from A31.rsrc_info") %>%
+    add_legacy_name("L231.UnlimitRsrcPrice") %>%
+    add_precursors("emissions/A31.rsrc_info") ->
+    L231.UnlimitRsrcPrice
+
+    L231.FinalDemand_urb %>%
+    add_title("Urban Processes Final Energy Demand") %>%
+    add_units("units") %>%
+    add_comments("Constants for base service, income elasticity, and aeei") %>%
+    add_legacy_name("L231.FinalDemand_urb") %>%
+    add_precursors("emissions/A_regions") ->
+    L231.FinalDemand_urb
+
+    L231.Supplysector_urb_ind %>%
+    add_title("Urban and Industrial Processes Supplysector Logit Info") %>%
+    add_units("Unitless") %>%
+    add_comments("A31.sector written to all regions") %>%
+    add_legacy_name("L231.Supplysector_urb_ind") %>%
+    add_precursors("emissions/A31.sector", "common/GCAM_region_names") ->
+    L231.Supplysector_urb_ind
+
+    L231.SubsectorLogit_urb_ind %>%
+    add_title("Urban and Industrial Processes Subsector Logit Info") %>%
+    add_units("Unitless") %>%
+    add_comments("A31.subsector_logit written to all regions") %>%
+    add_legacy_name("L231.SubsectorLogit_urb_ind") %>%
+    add_precursors("emissions/A31.subsector_logit", "common/GCAM_region_names") ->
+    L231.SubsectorLogit_urb_ind
+
+    if (exists("L231.SubsectorShrwt_urb_ind")){
```

Comment:
# This was a mistake in the old data system - the wrong name was assigned, but because it didn't exist, no error was thrown